subsys/net/lib/wifi_credentials/wifi_credentials.c

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <net/wifi_credentials.h>
+#include <sys/types.h>
+#include <stdlib.h>
+
+#define ENTRY_MAX_LEN 256
+
+
+/* SSIDs can contain any data - we handle them as NULL-terminated strings. */
+static struct wifi_credentials_ssid_entry entries[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES];
+static sys_slist_t wifi_credentials_entry_list = SYS_SLIST_STATIC_INIT(&bt_auth_info_cbs);
+
+LOG_MODULE_REGISTER(wifi_credentials, CONFIG_WIFI_CREDENTIALS_LOG_LEVEL);
+
+/**
+ * @brief Stores settings entry in flash
+ *
+ * @param idx credential index
+ * @param buf encoded settings entry
+ * @param buf_len length of buf
+ * @return 0 on success, -ERRNO on error
+ */
+int wifi_credentials_pickle_entry(size_t idx, const uint8_t *buf, size_t buf_len);
+
+/**
+ * @brief Deletes settings entry from flash
+ *
+ * @param idx credential index
+ * @return 0 on success, -ERRNO on error
+ */
+int wifi_credentials_delete_entry(size_t idx);
+
+/**
+ * @brief Loads settings entry from flash
+ *
+ * @param idx credential index
+ * @param buf encoded settings entry
+ * @param buf_len length of buf
+ * @return 0 on success, -ERRNO on error
+ */
+int wifi_credentials_unpickle_entry(size_t idx, uint8_t *buf, size_t buf_len);
```

Comment:
I'd drop `wifi` prefix from all private functions. I would also rename `entry` to `nv`. This brings more information in the function name. Following @jtguggedal suggestion these could be:

```
credentials_serialize_nv()
credentials_deserialize_nv()
credentials_delete_nv()
```

Reply:
now that these functions are moved to a separate file, we need the prefix to prevent name collisions